op-chain-ops/deployer/pipeline/opchain.go

```diff
@@ -91,13 +94,46 @@ func DeployOPChain(ctx context.Context, env *Env, artifactsFS foundry.StatDirFs,
 		DelayedWETHPermissionedGameProxyAddress:   dco.DelayedWETHPermissionedGameProxy,
 		DelayedWETHPermissionlessGameProxyAddress: dco.DelayedWETHPermissionlessGameProxy,
 	})
+
+	currentBlock, _ := env.L1Client.BlockNumber(ctx)
+	block, _ := env.L1Client.BlockByNumber(ctx, big.NewInt(int64(currentBlock)))
+	currentBlockHash := block.Hash()
+
+	// If any of the implementations addresses (excluding OpcmProxy) are empty,
+	// we need to set them using the addresses of the corresponding proxies.
```

Comment:
nit, since I interpreted this to mean we were setting the proxy address (feel free to tweak this suggestion)
```suggestion
	// we need to set them using the implementation address read from their corresponding proxy.
```